python/mxnet/rnn/rnn_cell.py

```diff
@@ -1068,3 +1069,353 @@ def unroll(self, length, inputs, begin_state=None, layout='NTC', merge_outputs=N
 
         states = [l_states, r_states]
         return outputs, states
+
+class ConvRNNCell(BaseRNNCell):
+    """Convolutional RNN cells
+
+    Parameters
+    ----------
+    input_shape : tuple of int
+        Shape of input in single timestep.
+    num_hidden : int
+        Number of units in output symbol.
+    h2h_kernel : tuple of int, default (3, 3)
+        Kernel of Convolution operator in state-to-state transitions.
+    h2h_pad : tuple of int, default (1, 1)
+        Pad of Convolution operator in state-to-state transitions.
```

Comment:
LGTM in general. Here we may be able to remove the h2h_pad as it is determined by h2h_kernel and h2h_dilate.